services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -113,4 +120,104 @@ class SwitchRulesSpec extends BaseSpecification {
             northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.size() == defaultRules.size()
         }
     }
+
+    @Unroll
+    def "Able to delete #data.description rules from a switch"() {
+        given: "A switch with some flow rules installed"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northboundService.getFlowStatus(flow.id).status == FlowState.UP &&
+                    northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.size() ==
+                    defaultRules.size() + flowRulesCount
+        }
+
+        when: "Delete #data.description rules from the switch"
+        def deletedRules = northboundService.deleteSwitchRules(srcSwitch.dpId, data.deleteRulesAction)
+
+        then: "#data.description.capitalize() rules are really deleted"
+        deletedRules.size() == data.rulesDeleted
+        Wrappers.wait(WAIT_OFFSET) {
+            northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.size() == data.rulesRemained
+        }
+
+        and: "Delete the flow"
+        northboundService.deleteFlow(flow.id)
+
+        and: "Install default rules if necessary"
+        if (data.rulesRemained == 0) {
+            northboundService.installSwitchRules(srcSwitch.dpId, InstallRulesAction.INSTALL_DEFAULTS)
+            Wrappers.wait(WAIT_OFFSET) {
+                northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.size() == defaultRules.size()
+            }
+        }
+
+        where:
+        data << [[description      : "non-default",
+                  deleteRulesAction: DeleteRulesAction.IGNORE_DEFAULTS,
+                  rulesDeleted     : flowRulesCount,
+                  rulesRemained    : defaultRules.size()
+                 ],
+                 [description      : "all",
+                  deleteRulesAction: DeleteRulesAction.DROP_ALL,
+                  rulesDeleted     : defaultRules.size() + flowRulesCount,
+                  rulesRemained    : 0
+                 ]
+        ]
+    }
+
+    def "Able to synchronize rules on a switch (install missing rules)"() {
+        given: "A switch with missing rules"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        northboundService.addFlow(flow)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northboundService.getFlowStatus(flow.id).status == FlowState.UP &&
+                    northboundService.getSwitchRules(srcSwitch.dpId).flowEntries.size() ==
```

Comment:
better to do 2 separate assert statements instead of using `&&`. It will give you a more clear output in case of failure